### PR TITLE
codereview

### PR DESCRIPTION
**Pull request by martingerlach**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,3 @@
 # wn
+
+## what is this tool doing?
```
Comment (martingerlach) [inline README.md:3]: Maybe we can add a note where the tool is hosted, what it is doing?

File: client/index.html [hunk 2]
```diff
@@ -1 +1,2 @@
...
...rn e}var n={},o={1:0},u=[];function l(t){if(n[t])return n[t].exports;var r=n[t]={i:t,l:!1,exports:{}};return e[t].call(r.exports,r,r.exports,l),r.l=!0,r.exports}l.m=e,l.c=n,l.d=function(e,t,r){l.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:r})},l.r=function(e){"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},l.t=function(e,t){if(1&t&&(e=l(e)),8&t)return e;if(4&t&&"object"==typeof e&&e&&e.__esModule)return e;var r=Object.create(null);if(l.r(r),Object.defineProperty(r,"default",{enumerable:!0,value:e}),2&t&&"string"!=typeof e)for(var n in e)l.d(r,n,function(t){return e[t]}.bind(null,n));return r},l.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return l.d(t,"a",t),t},l.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},l.p="/";var i=this["webpackJsonpwn-client"]=this["webpackJsonpwn-client"]||[],f=i.push.bind(i);i.push=t,i=i.slice();for(var a=0;a<i.length;a++)t(i[a]);var c=f;r()}([])</script><script src="/static/js/2.812377b0.chunk.js"></script><script src="/static/js/main.35aa9034.chunk.js"></script></body></html>
```
Comment (martingerlach) [inline client/index.html:2]: This seems to be working (it looks great!) but I couldnt parse the content of the html

File: generate_db.py [hunk 3]
```diff
@@ -3,11 +3,11 @@
 import sys
 import os
 
-language = sys.argv[1]
+language = sys.argv[1] # the expected input is enwiki
```
Comment (martingerlach) [inline generate_db.py:6]: is the expected input en or enwiki or en.wikipedia?

File: generate_db.py [hunk 4]
```diff
@@ -3,11 +3,11 @@
...
 month = '2021-01'
 dump_dir = f"/public/dumps/public/other/clickstream/{month}/"
 clickstream_fn = f'clickstream-{language}-{month}.tsv.gz'
-database = os.path.expanduser(f'~/www/python/src/db/clickstream_{language[:2]}.db')
+database = os.path.expanduser(f'~/www/python/src/db/clickstream_{language[:2]}.db') # language.replace("wiki","")
```
Comment (martingerlach) [inline generate_db.py:10]: maybe the current approach is even better, but an alternative is to replace the wiki

**Review by martingerlach — COMMENTED** [review]
The code looks great.
I now understand how the caching makes sense when using arguments destinations/sources or limit that we still only make one query, great idea : )

File: README.md (hunk 1 above)
Comment (mnzpk) [reply README.md:3]: You're right! I've been a little busy making changes to the interface but expect this to look a little less sad in a day or two :)

File: client/index.html (hunk 2 above)
Comment (mnzpk) [reply client/index.html:2]: This is is the minified HTML file that was generated when I built my react code. The client folder only contains static files right now. I'll push the react code that I've been editing as soon as I'm done finalizing a few changes.

File: generate_db.py (hunk 3 above)
Comment (mnzpk) [reply generate_db.py:6]: The expected input is enwiki but I think en would be better(?) as the behavior right now is a little confusing.

File: generate_db.py (hunk 4 above)
Comment (mnzpk) [reply generate_db.py:10]: Replacing the wiki looks more readable. I'll fix it. Thanks!

**Pull request closed without merging**